packages/multicast_dns/lib/src/packet.dart

```diff
@@ -162,7 +162,16 @@ _FQDNReadResult _readFQDN(
         final Uint8List partBytes =
             Uint8List.view(data.buffer, offset, partLength);
         offset += partLength;
-        parts.add(utf8.decode(partBytes));
+
+        try {
+          parts.add(utf8.decode(partBytes));
+        } on FormatException {
+          throw MDnsDecodeException(offset,
+              detailError: 'Bad UTF-8 encoding found while decoding string. '
+                  'The Flutter team would greatly appreciate if you could file a bug or leave a'
```

Comment:
This message is too specific for a general purpose package. the flutter_tool should add the flutter specific phrasing if it catches this error

Reply:
In that case, we should probably either create a new type of exception or add an error code field to this one.  I would probably lean towards a new type of exception that derives from MDnsDecodeException, e.g. MDnsUtf8DecodeException that still has the offset as well as the byte list.  That way the tool could do a catch on it and do whatever it thinks is necessary with the byte array.

Reply:
SGTM, I also lean toward a new type of exception.